FBSnapshotTestCaseDemo/FBSnapshotTestCaseDemoTests/FBSnapshotTestCaseDeviceAgnosticTests.m

```diff
@@ -0,0 +1,34 @@
+/*
+ *  Copyright (c) 2013, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import <FBSnapshotTestCase/FBSnapshotTestCase.h>
+
+@interface FBSnapshotTestCaseDeviceAgnosticTests : FBSnapshotTestCase
+
+@end
+
+@implementation FBSnapshotTestCaseDeviceAgnosticTests
+
+- (void)setUp
+{
+    [super setUp];
+    self.deviceAgnostic = YES;
+    self.recordMode = NO;
+}
+
+- (void)testViewSnapshot
+{
+    UIView *redView = [[UIView alloc] initWithFrame:CGRectMake(0, 0, 40, 40)];
+    redView.backgroundColor = [UIColor redColor];
+    FBSnapshotVerifyView(redView, nil);
```

Comment:
@nscoding yeah, sorry I didn't respond, but better late than never:

> The build script that I added on travis is building and testing both iPhone 5 and iPhone 6, so we have to record the image for both of these devices.

I already figured that out when Travis failed and downloaded/recorded the proper screenshots.

> testFailedImageWithDeviceAgnosticShouldHaveModelOnName. These tests don't have an application associated with, so we hit the assertion. 

Those tests are in the Demo project, which does have a host application (see above screenshot).  I _think_ the problem was that `UIImage+Snapshot` was causing issues when installing a "dummy" window when given a view without a window.  This somehow caused the `keyWindow` to return `nil` when device-agnostic tests were run—which I determined after seeing each test pass in isolation.  As a result, I've modified `UIImage+Snapshot` to use the application's `keyWindow` if the given view doesn't have a window (or if it isn't a window itself).  Let me know if you think this is acceptable.

> Maybe we can make the window a parameter and simply write tests to make sure that FBDeviceAgnosticNormalizedFileName is working as expected.

That could be useful for verifying the formatting of test names (to prevent unintended regressions and thus break tests for users), but doesn't address the problems targeted by this PR.